src/Symfony/Component/Form/Extension/Validator/ViolationMapper/ViolationMapper.php

```diff
@@ -124,6 +126,17 @@ public function mapViolation(ConstraintViolation $violation, FormInterface $form
 
         // Only add the error if the form is synchronized
         if ($this->acceptsErrors($scope)) {
+            if (File::TOO_LARGE_ERROR === $violation->getCode()) {
```

Comment:
a better check could be `$violation->getCause() instanceof File` to account for the different error codes

Reply:
I don't think so. If there are multiple file upload files and one of them fails with another check of the `File` constraint we could potentially remove an error here that we would have to keep.